qemu/tests/virtio_fs_multi_vms.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import os
+
+from virttest import error_context
+from provider.virtiofs_utils import run_dd_test
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test to virtio-fs with the multiple VMs and virtiofs daemons.
+    Steps:
+        1. Create shared directories on the host.
+        2. Run virtiofs daemons on the host.
+        3. Boot guests on the host with virtiofs options.
+        4. Log into guest then mount the virtiofs targets.
+        5. Generate files on the mount points inside guests.
+        6. Compare the md5 between the host and the guest.
+        7. Compare the md5 among guests if multiple virtiofs
+           daemons share the source.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    def create_dest_dir(vm, session, fs_dest):
+        error_context.context("%s: Create a destination directory %s "
+                              "inside guest." % (vm, fs_dest), logging.info)
+        session.cmd(cmd_create_dest_dir % fs_dest)
+
+    def mount_target(vm, session, fs_target, fs_dest):
+        error_context.context("%s: Mount the virtiofs target %s to %s inside guest."
+                              % (vm, fs_target, fs_dest), logging.info)
+        session.cmd(cmd_mount_target % (fs_target, fs_dest))
+
+    def umount_target(vm, session, fs_target, fs_dest):
+        error_context.context("%s: Umount the viriofs target %s." % (vm, fs_target),
+                              logging.info)
+        session.cmd(cmd_umount_target % fs_dest)
+
+    def compare_vms_shared_md5():
```

Comment:
same here

Reply:
Sorry this function did have the universality, it just availabe in this scenarios.